File: frontend/types/analytics.ts
export interface DepartmentAnalytics {
  name: string;
  count: number;
}

export interface AnalyticsData {
   employees: number;
  projects: number;
  seats: number;
  occupied: number;
  available: number;
  reserved: number;
  maintenance: number;
  pending_joiners: number;
  utilization: number;
}

File: backend/app/schemas/analytics.py
from pydantic import BaseModel


class DashboardSummary(BaseModel):
    employees: int
    seats: int
    occupied: int
    available: int
    utilization: float


class DepartmentAnalytics(BaseModel):
    department: str
    employees: int


class ProjectAnalytics(BaseModel):
    project: str
    employees: int